StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Saved Payment Method Screen/Vertical Saved Payment Method Screen/PaymentMethodRowButton.swift

```diff
@@ -14,34 +14,29 @@ import UIKit
 protocol PaymentMethodRowButtonDelegate: AnyObject {
     func didSelectButton(_ button: PaymentMethodRowButton, with paymentMethod: STPPaymentMethod)
     func didSelectRemoveButton(_ button: PaymentMethodRowButton, with paymentMethod: STPPaymentMethod)
-    func didSelectEditButton(_ button: PaymentMethodRowButton, with paymentMethod: STPPaymentMethod)
+    func didSelectUpdateButton(_ button: PaymentMethodRowButton, with paymentMethod: STPPaymentMethod)
 }
 
 // TODO: Make this use RowButton internally
 final class PaymentMethodRowButton: UIView {
 
-    struct ViewModel {
-        let appearance: PaymentSheet.Appearance
-        let paymentMethod: STPPaymentMethod
-        // TODO(porter) Add can remove and can update
-    }
-
     enum State {
         case selected
         case unselected
-        case editing
+        case editing(allowsRemoval: Bool, allowsUpdating: Bool)
     }
 
     // MARK: Internal properties
     var state: State = .unselected {
         didSet {
             previousState = oldValue
 
-            selectionTapGesture.isEnabled = !isEditing
-            shadowRoundedRect.isSelected = isSelected
+            // Any gesture recognizers on the rowButton will intercept taps on the update and remove buttons, disable them if in edit mode
```

Comment:
Can you try putting the gesture recognizer on the shadowed rounded rect in `RowButton`?

Reply:
Good idea! Yeah that works, one side-effect is that we need to disable interaction on the labels and image since they sit above the rounded rect, but this seems reasonable to me.

Alternatively we could add those views as subviews on the rounded rect.